compiler/mx.compiler/mx_jaotc.py

```diff
@@ -0,0 +1,127 @@
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+# ----------------------------------------------------------------------------------------------------
+
+import subprocess
+import tempfile
+from argparse import ArgumentParser
+
+import mx
+import mx_compiler
+from mx_gate import Task
+
+jdk = mx.get_jdk(tag='default')
+
+
+def run_jaotc(args, cwd=None):
+    if jdk.javaCompliance < '11':
+        mx.abort('jaotc command is only available if JAVA_HOME is JDK 11 or later')
+    jaotc_entry = mx_compiler.JVMCIClasspathEntry('JAOTC')
+    jvmci_classpath_adjusted = False
+    if jaotc_entry not in mx_compiler._jvmci_classpath:
+        mx_compiler.add_jvmci_classpath_entry(jaotc_entry)
+        jvmci_classpath_adjusted = True
+
+    vm_args = [a[2:] for a in args if a.startswith('-J')]
+    args = [a for a in args if not a.startswith('-J')]
+
+    try:
+        mx_compiler.run_vm(['--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.sparc=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=jdk.internal.vm.compiler,jdk.aot',
+                            '--add-exports=jdk.internal.vm.ci/jdk.vm.ci.sparc=jdk.internal.vm.compiler,jdk.aot',
+                            '-XX:+CalculateClassFingerprint'] + vm_args + ['-m', 'jdk.aot/jdk.tools.jaotc.Main'] + args,
+                           cwd=cwd)
+    finally:
+        if jvmci_classpath_adjusted:
+            mx_compiler._jvmci_classpath.remove(jaotc_entry)
+
+
+def jaotc_gate_runner(tasks):
+    with Task('jaotc', tasks, tags=['jaotc', 'fulltest']) as t:
+        if t: jaotc_tests([])
+
+
+def jaotc_tests(args):
+    parser = ArgumentParser(prog='mx jaotc-tests')
+    parser.add_argument('test', help='test to run', nargs='*')
```

Comment:
Should the help text be `tests to run (omit to run all tests)`?
Also, since the set of tests is closed, you could add `choices=['HelloWorld']`.